lib/features/debug/presentation/screens/ui_kit/ui_kit_wm.dart

```diff
@@ -0,0 +1,106 @@
+import 'package:elementary/elementary.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_template/common/mixin/localization_mixin.dart';
+import 'package:flutter_template/common/mixin/theme_mixin.dart';
+import 'package:flutter_template/features/app/di/app_scope.dart';
+import 'package:flutter_template/features/debug/presentation/screens/ui_kit/ui_kit_model.dart';
+import 'package:flutter_template/features/debug/presentation/screens/ui_kit/ui_kit_screen.dart';
+import 'package:provider/provider.dart';
+
+/// Factory for [UiKitWM].
+UiKitWM uiKitScreenWMFactory(BuildContext context) {
+  final appScope = context.read<IAppScope>();
+
+  final model = UiKitModel(logWriter: appScope.logger);
+
+  return UiKitWM(model);
+}
+
+/// Interface for [UiKitWM].
+abstract class IUiKitWM with ILocalizationMixin, ThemeIModelMixin implements IWidgetModel {
+  /// Theme switching callback.
+  void switchTheme();
+
+  /// Callback of pressing the primary button.
+  void onPrimaryButtonPressed();
+
+  /// Callback of pressing the secondary button.
+  void onSecondaryButtonPressed();
+
+  /// Callback of pressing the tertiary button.
+  void onTertiaryButtonPressed();
+
+  /// Callback of pressing the tetradic button.
+  void onTetradicButtonPressed();
+
+  /// Callback of pressing the danger snack button.
+  void onDangerSnackButtonPressed();
+
+  /// Callback of pressing the positive snack button.
+  void onPositiveSnackButtonPressed();
+}
+
+/// {@template ui_kit_widget_model.class}
+/// [WidgetModel] for [UiKitScreen].
+/// {@endtemplate}
+class UiKitWM extends WidgetModel<UiKitScreen, UiKitModel> with LocalizationMixin, ThemeWMMixin implements IUiKitWM {
+  late final ScaffoldMessengerState _scaffoldMessenger;
```

Comment:
I don't think that we need `late final` variable. We can inject `ScaffoldMessengerState` through `UiKitWM` constructor.